syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package transport_catalogue_proto;

message Stop {
	uint32 id = 1;
	string name = 2;
	double latitude = 3;
	double longitude = 4;
}

message Bus {
	uint32 id = 1;
	string name = 2;
	repeated uint32 stops = 3;
	bool is_roundtrip = 4;
}

message Distance {
	uint32 stop_from = 1;
	uint32 stop_to = 2;
	uint32 distance = 3;
}

message CatalogueData {
	repeated Stop stops = 1;
	repeated Bus buses = 2;
	repeated Distance distance_between_stops = 3;
}

message ProcessingData {
	CatalogueData catalogue = 1;
	RenderSettingsData render_settings = 2;
	RouterData router = 3;
}